index.js

```diff
@@ -0,0 +1,348 @@
+'use strict';
+
+const Scm = require('screwdriver-scm-base');
+const async = require('async');
+const hoek = require('hoek');
+
+class ScmRouter extends Scm {
+    /**
+     * Constructs a router for different scm strategies
+     * @method constructor
+     * @param  {Object}         config                             Object with scms and ecosystem
+     * @param  {Object}         [config.ecosystem]                 Optional object with ecosystem values
+     * @param  {Object}         config.scms                        Array of scms to load or a single scm object
+     * @param  {String}         config.scms[x].plugin              Name of the scm NPM module to load
+     * @param  {String}         config.scms[x].config              Configuration to construct the module with
+     * @param  {String}         config.scms[x].config.displayName  Nickname to displaoy of the scm
+     * @return {ScmRouter}
+     */
+    constructor(config = {}) {
+        const ecosystem = config.ecosystem;
+        const scmsConfig = config.scms;
+
+        super();
+
+        this.scms = {};
+
+        if (typeof scmsConfig === 'object') {
+            if (!(Array.isArray(scmsConfig))) {
+                throw new Error('No scm config passed in.');
+            }
+            scmsConfig.forEach((scm) => {
+                if (typeof scm.config === 'object' && typeof scm.config.displayName === 'string') {
+                    const options = hoek.applyToDefaults({ ecosystem }, scm.config);  // Add ecosystem to scm options
+
+                    this.loadPlugin(scm.plugin, options);
+                } else {
+                    throw new Error(`Display name not specified for ${scm.plugin} scm plugin`);
+                }
+            });
+        }
+
+        if (Object.keys(this.scms).length === 0) {
+            throw new Error('No scm config passed in.');
+        }
+    }
+
+    /**
+     * load scm module
+     * @method loadPlugin
+     * @param  {String}         plugin       load plugin name
+     * @param  {Object}         options      settings for scm module
+     */
+    loadPlugin(plugin, options) {
+        if (plugin === 'router') {
+            console.warn('The plugin of scm-router can not be specified for scms setting');
+
+            return;
+        }
+
+        // eslint-disable-next-line global-require, import/no-dynamic-require
+        const ScmPlugin = require(`screwdriver-scm-${plugin}`);
```

Comment:
I don't see any validation to ensure that the `plugin` passed in is valid. If an invalid `plugin` is passed in, this would cause an error and it's not being handled anywhere.